ui/src/pages/Diphtheria.jsx

```diff
@@ -0,0 +1,443 @@
+      let visit = window.localStorage.getItem("currentPatient") ?? null;
+      visit = JSON.parse(visit) ?? null;
+      if (visit) {
+        getDiphtheriaVaccinationEncounters(visit.id);
+      }
+    }, []);
+  
+    let getEncounterObservations = async (encounter) => {
+      setObservations([]);
+      handleOpen();
+      let observations = await (
+        await FhirApi({
+          url: `/crud/observations?encounter=${encounter}`,
+        })
+      ).data;
+      setObservations(observations.observations);
+      return;
+    };
+  
+    let getDiphtheriaVaccinationEncounters = async (patientId) => {
+      setLoading(true);
+      let encounters = await (
+        await FhirApi({
+          url: `/crud/encounters?patient=${patientId}&encounterCode=${"DIPHTHERIA_VACCINATION"}`,
+        })
+      ).data;
+      console.log(encounters);
+      setDiphtheriaVaccinationEncounters(encounters.encounters);
+      setLoading(false);
+      return;
+    };
+    let patientName = visit?.name;
+    
+    let saveDiphtheriaVaccination = async (values) => {
+      //get current patient
+      if (!visit) {
+        prompt(
+          "No patient visit not been initiated. To start a visit, Select a patient in the Patient's list"
+        );
+        return;
+      }
+      let patient = visit.id;
+      
+      
+      try {
+        //create Encounter
+        let encounter = await createEncounter(patient, "DIPHTHERIA_VACCINATION");
+
+        let immunizationDetails={
+          patientId:patient,
+          encounterId:encounter.id,
+          practitionerId,
+          name:'diptheria',
+          immunizationDate:values.dateGiven,
+          lotNumber:values.lotNumber,
+          expiryDate:values.dateOfExpiry,
+          site:'leftThigh',
+          route:'injectionIntramuscular',
+          dosage:0.5,
+          unit:'mls'
+        }
+        let immunization=await createImmunization(immunizationDetails)
+        
+
+        let appointmentDetails={
+          serviceCategory:1,
+          reason:3,
+          description:'Routine diphtheria vaccination appointment',
+          nextVisit:values.dateOfNextVisit,
+          patientId:patient,
+          patientName,
+          practitionerId,
+          practitionerName:names,
+        }
+
+        let appointment=diphtheriaVaccinationEncounters.length < 2 && (await createAppointment(appointmentDetails))
+
+  
+        // Create and Post Observations
+        let res = await (
+          await fetch(`${apiHost}/crud/observations`, {
+            method: "POST",
+            body: JSON.stringify({
+              patientId: patient,
+              encounterId: encounter.id,
+              // observations: values.ret,
+            }),
+            headers: { "Content-Type": "application/json" },
+          })
+        ).json();
+        
+  
+        if (immunization.status  === "success" || (diphtheriaVaccinationEncounters.length < 2 && appointment.status === "success")) {
```

Comment:
I am of the idea that we separate the checking of the response status. Here is how we do it.
- We submit the immunization details
- Based on its success, we conditionally create the appointments.
**NOTE :** Failure of any of the operations should be relayed to the user.